src/pyhf/tensor/numpy_backend.py

```diff
@@ -205,9 +206,12 @@ def conditional(
         """
         return true_callable() if predicate else false_callable()
 
-    def tolist(self, tensor_in: Tensor[T] | list[T]) -> list[T]:
+    def tolist(
+        self, tensor_in: Tensor[T] | list[T]
+    ) -> int | float | complex | list[T] | list[Any]:
         try:
-            return tensor_in.tolist()  # type: ignore[union-attr,no-any-return]
+            result = tensor_in.tolist()  # type: ignore[union-attr]
+            return cast(Union[int, float, complex, list[T], list[Any]], result)
```

Comment:
PR https://github.com/scikit-hep/pyhf/pull/2625 is the required/sufficient of this PR to drop Python 3.8. What is the need for the `pyhf.tensor.manager` imports in this PR along with the casting working in `tolist`?

If this is a general improvement to typing that is needed to make something easier then we should factor that out from a Python support change.

Reply:
>  but we still have many issues to deal with that are going to be hard to disentangle and not worth the time to do so when we're just trying to fell-swoop fix all issues

@kratsg I don't think I understand the issues that exist for typing here, which is my fault. Can you elaborate them in an Issue so that we can get them addressed, regardless of if we get them into the `v0.8.0` release?